### ISSUE TITLE
How could I use NTXent loss to perform a self-supervised representation learning without labels?

### ISSUE THREAD

**Issue by HilbertHuangHitomi** [KevinMusgrave/pytorch-metric-learning#335]
Suppose that we have a set of points as the input `[batch_size, input_dim]`. 
After some augmentations, we have several different views from each point, and now the shape becomes `[batch_size, n_views, input_dim]`. 
Using an encoder, we encode all points into a latent space `[batch_size, n_views, latent_dim]`.

Now I would like to define the NTXent loss on the latent space to train the encoder. The input shape is `[batch_size, n_views, latent_dim]`. We won't have explicit labels to indicate the positive/negative pairs.

So, is there any efficient means to calculate the loss for me without using a for loop in the training loop like `SupContrast` in `SupContrast`, please?

**Comment by KevinMusgrave** [comment]
Flatten your batch so that it has shape ```(batch_size * n_views, latent_dim)```. 

Then make labels like this:
```python
labels = torch.arange(batch_size).repeat_interleave(n_views)
```
For example, if ```batch_size=5``` and ```n_views=3``` then ```labels=tensor([0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4])```.

Now pass in the embeddings and labels into NTXentLoss:

```python
loss_fn = NTXentLoss()

loss = loss_fn(embeddings, labels)
```

**Comment by HilbertHuangHitomi** [comment]
Thank you very much! >ω<
